### PR TITLE
tests/nested/manual/core20-save: verify handling of ubuntu-save with different system variants 

### PR DESCRIPTION
**Pull request by bboozzoo**
Verify that ubuntu-save is handled properly in 3 possible system setups:

- TPM & secure boot
- ubuntu-save in the gadget, but no TPM or secure boot
- no ubuntu-save in the gadget, no TPM or secure boot

### CODE REVIEW

**Review by mvo5 — APPROVED** [review]
Nice, thank you!

**Comment by bboozzoo** [comment]
Hmm, google-nested:ubuntu-20.04-64:tests/nested/manual/core20-save:plain somehow failed here, but locally it is passing:
```
2020-11-13 13:05:05 Preparing google-nested:ubuntu-20.04-64:tests/nested/manual/...
2020-11-13 13:05:44 Preparing google-nested:ubuntu-20.04-64:tests/nested/manual/core20-save:plain...
2020-11-13 13:15:01 Executing google-nested:ubuntu-20.04-64:tests/nested/manual/core20-save:plain (1/1)...
2020-11-13 13:15:52 Restoring google-nested:ubuntu-20.04-64:tests/nested/manual/core20-save:plain...
2020-11-13 13:15:55 Restoring google-nested:ubuntu-20.04-64:tests/nested/manual/...
2020-11-13 13:16:10 Restoring google-nested:ubuntu-20.04-64...
2020-11-13 13:16:11 Discarding google-nested:ubuntu-20.04-64 (nov131155-474309)...
2020-11-13 13:16:12 Successful tasks: 1
2020-11-13 13:16:12 Aborted tasks: 0
```

File: tests/nested/manual/core20-save/task.yaml [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  NESTED_ADD_UBUNTU_SAVE/plainwithsave: true
+
+  # no ubuntu-save, no TPM, no secure boot
+  NESTED_ENABLE_TPM/plain: false
+  NESTED_ENABLE_SECURE_BOOT/plain: false
+  NESTED_ADD_UBUNTU_SAVE/plain: false
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
```
Comment (sergiocazzolato) [inline tests/nested/manual/core20-save/task.yaml:28]: No need to source this

Comment (bboozzoo) [reply tests/nested/manual/core20-save/task.yaml:28]: Ahh right, we got the *-state tools now.

**Review by sergiocazzolato — APPROVED** [review]
Nice, thanks for the test, minor comment inline

File: tests/nested/manual/core20-save/task.yaml [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    "$TESTSTOOLS"/nested-state build-image core
+    # NESTED_ENABLE_TPM is used by create-vm
+    "$TESTSTOOLS"/nested-state create-vm core
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    if [ "$NESTED_ADD_UBUNTU_SAVE" != "true" ] && [ "$NESTED_ENABLE_TPM" != "true" ]; then
+        # no ubuntu-save, make sure the system is up
```
Comment (anonymouse64) [inline tests/nested/manual/core20-save/task.yaml:40]:
```suggestion
        # no ubuntu-save, make sure one isn't mounted
```

File: tests/nested/manual/core20-save/task.yaml [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        save_out="$(nested_exec "df -B1 /run/mnt/ubuntu-save | tail -1")"
+        if [ "$NESTED_ENABLE_TPM" = "true" ]; then
+            # when encrypted, ubuntu-save is mounted from a dm device
+            echo "$save_out" | MATCH '^/dev/mapper/ubuntu-save-[0-9a-z-]+\s+'
+        else
+            # when plain ubuntu-save is mounted from a regular block device
+            echo "$save_out" | MATCH '^/dev/vda[0-9]\s+'
+        fi
+        save_size="$(echo "$save_out" | awk '{print $4}')"
+        echo "check  there is at least 6MB of free space available on ubuntu-save"
```
Comment (anonymouse64) [inline tests/nested/manual/core20-save/task.yaml:57]:
```suggestion
        echo "check that there is at least 6MB of free space available on ubuntu-save"
```

**Review by anonymouse64 — APPROVED** [review]
LGTM, a couple little nitpicks

**Comment by bboozzoo** [comment]
Magically passed after adding NESTED_IMAGE_ID. I'll try closing and reopening to get a full spread run.

**Comment by bboozzoo** [comment]
The test succeeeded 2 since I pushed NESTED_IMAGE_ID. I'm going to land this now. If problems arise, we'll switch the test to manual.

**Pull request merged**